diff --git a/dist/index.js b/dist/index.js
index badee5803289701abe5c68072e2c5b4df5eae731..37f9ba50386036fccd3eaa444b701fd5dec60b93 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -8654,7 +8654,7 @@ var WalletAccount = class _WalletAccount extends Account {
     super(providerOrOptions, address, "", cairoVersion);
     this.walletProvider = walletProvider;
     this.walletProvider.on("accountsChanged", (res) => {
-      if (!res) return;
+      if (!res || res.length === 0) return;
       this.address = res[0].toLowerCase();
     });
     this.walletProvider.on("networkChanged", (res) => {
diff --git a/dist/index.mjs b/dist/index.mjs
index cd1c7da0b13a5e529db6ced2cba37f1c2eb11f60..1e6f020964285ba5f006c77a44fb8bc289273466 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -8533,7 +8533,7 @@ var WalletAccount = class _WalletAccount extends Account {
     super(providerOrOptions, address, "", cairoVersion);
     this.walletProvider = walletProvider;
     this.walletProvider.on("accountsChanged", (res) => {
-      if (!res) return;
+      if (!res || res.length === 0) return;
       this.address = res[0].toLowerCase();
     });
     this.walletProvider.on("networkChanged", (res) => {
